src/libmongoc/src/mongoc/mongoc-server-monitor.c

```diff
@@ -326,10 +326,12 @@ _server_monitor_polling_ismaster (mongoc_server_monitor_t *server_monitor,
                                   bson_error_t *error)
 {
    bson_t cmd;
+   const bson_t *hello;
    bool ret;
 
-   bson_init (&cmd);
-   bson_append_int32 (&cmd, "isMaster", 8, 1);
+   hello = _mongoc_topology_scanner_get_hello_cmd (server_monitor->topology->scanner);
```

Comment:
Renaming this was a good idea. And reducing the duplicate generation of the hello command will make it easier to update when we need to deal with helloOk.

There's no need to change it now. But in CDRIVER-3682 I intend to remove the dependency of the server monitor on the topology scanner to keep then separate. The topology scanner is exclusive to single-threaded scanning, and the server monitor is exclusive to multi-threaded. I can make a note to move the handshake generation outside of the topology scanner as part of CDRIVER-3682 if that sounds ok to you.

Reply:
That sounds good. Thanks!